### ISSUE TITLE
nexus-inference: MLP enhancements — LayerNorm and optional bias

### ISSUE THREAD

**Issue by Abso1ut3Zer0** [Abso1ut3Zer0/nexus#349]
## Context

`MlpF32`/`MlpF64` currently support arbitrary layer sizes, 9 activation functions, and bias vectors. PyTorch's equivalent (`nn.Sequential` of `nn.Linear` + normalization + activation) offers two additional features commonly used in production models.

## Current offering

- Arbitrary hidden layer sizes via builder
- Per-layer activation: Relu, Tanh, Sigmoid, Gelu, Swish, Elu, LeakyRelu, Identity, SiLU
- Bias on every layer (always present)
- SIMD-tiled matmul (AVX2/AVX-512)
- `predict` / `predict_into` API

### 1. LayerNorm / BatchNorm between layers

PyTorch production MLPs frequently use `nn.LayerNorm` between linear layers:

```python
model = nn.Sequential(
    nn.Linear(8, 32),
    nn.LayerNorm(32),
    nn.ReLU(),
    nn.Linear(32, 1),
)
```

LayerNorm at inference is: `y = gamma * (x - mean) / sqrt(var + eps) + beta`, where `gamma` and `beta` are learned affine parameters stored in the state_dict. This is a per-element operation — no cross-sample statistics needed at inference time (unlike BatchNorm which uses running mean/var).

**Recommendation:** Support LayerNorm first (inference-friendly, no running statistics). BatchNorm is also straightforward at inference time (uses frozen running_mean/running_var from training).

### 2. Optional bias (`bias=False`)

Some architectures train linear layers without bias:

```python
nn.Linear(8, 32, bias=False)  # no bias term
```

Currently our MLP always adds bias. If a user trains with `bias=False` and loads those weights, inference will be wrong (bias tensor missing or zeroed when it shouldn't be applied).

**Recommendation:** Per-layer bias flag, or detect absence of bias tensor in safetensors and skip the add.

## Limitation for users

**Until this is implemented:** Users must ensure their PyTorch models use bias on all linear layers and do not include LayerNorm/BatchNorm between layers. Models violating these constraints will either fail to load or produce incorrect inference results.

## Priority

Enhancement — not blocking safetensors loader or test infrastructure work. These can be added incrementally.
